### ISSUE TITLE
Unhandled exceptions in Light mode

### ISSUE THREAD

**Issue by cburgdorf** [ethereum/py-evm#876]
### What is wrong?

This appeared when running Trinity with `--light`

```
 WARNING  06-06 12:07:02  lightchain  LESAnnouncementProcessingError('Too many timeouts when fetching headers from LESPeer <Node(0x374a@51.255.168.86)>',)
   ERROR  06-06 12:07:02  base_events  Task exception was never retrieved
future: <Task finished coro=<LightPeerChain._process_announcements() done, defined at /home/cburgdorf/Documents/hacking/ef/py-evm/p2p/lightchain.py:141> exception=RuntimeError('Cannot cancel a service that has not been started',)>
Traceback (most recent call last):
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/p2p/lightchain.py", line 217, in process_announcement
    batch = await self.fetch_headers(start_block, peer)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/p2p/lightchain.py", line 172, in fetch_headers
    raise TooManyTimeouts()
p2p.exceptions.TooManyTimeouts

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/p2p/lightchain.py", line 150, in _process_announcements
    await self.process_announcement(peer, head_info)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/p2p/lightchain.py", line 220, in process_announcement
    "Too many timeouts when fetching headers from {}".format(peer))
p2p.exceptions.LESAnnouncementProcessingError: Too many timeouts when fetching headers from LESPeer <Node(0x374a@51.255.168.86)>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/p2p/lightchain.py", line 157, in _process_announcements
    await self.drop_peer(peer)
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/p2p/lightchain.py", line 82, in drop_peer
    await peer.cancel()
  File "/home/cburgdorf/Documents/hacking/ef/py-evm/p2p/service.py", line 93, in cancel
    raise RuntimeError("Cannot cancel a service that has not been started")
RuntimeError: Cannot cancel a service that has not been started
    INFO  06-06 12:07:22        peer  Connected peers: 0 inbound, 0 outbound
```

### How can it be fixed

:hear_no_evil:

**Comment by carver** [comment]
This might get fixed by the bonus change in #905

**Comment by gsalgado** [comment]
That code has been removed as we're now using `BaseHeaderChainSyncer` for light syncing, so this should no longer happen
